client/me/developer/dev-survey-notice.tsx

```diff
@@ -0,0 +1,78 @@
+import { Gridicon } from '@automattic/components';
+import { Button } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import surveyImage from 'calypso/assets/images/illustrations/developer-survey.svg';
+
+type DeveloperSurveyNoticeProps = {
+	onClose: ( remindTimeInSeconds: number, buttonName: string ) => void;
+	onSurveyClick: () => void;
+	localeSlug: string;
+};
+
+const ONE_DAY_IN_SECONDS = 24 * 60 * 60;
+const ONE_YEAR_IN_SECONDS = 365 * ONE_DAY_IN_SECONDS;
+
+export const DeveloperSurveyNotice = ( {
+	onClose,
+	onSurveyClick,
+	localeSlug,
+}: DeveloperSurveyNoticeProps ) => {
+	const translate = useTranslate();
+
+	const href =
+		localeSlug === 'es'
+			? 'https://wordpressdotcom.survey.fm/developer-survey-es'
+			: 'https://wordpressdotcom.survey.fm/developer-survey';
+
+	return (
+		<div className="developer-survey-notice">
+			<Button
+				className="developer-survey-notice__backdrop"
+				onClick={ () => onClose( ONE_DAY_IN_SECONDS, 'backdrop' ) }
+			/>
+			<div className="developer-survey-notice__popup">
+				<div className="developer-survey-notice__popup-head">
+					<div className="developer-survey-notice__popup-head-title">
+						{ translate( 'Developer Survey' ) }
+					</div>
+					<Button
+						onClick={ () => onClose( ONE_YEAR_IN_SECONDS, 'close-button' ) }
+						className="developer-survey-notice__popup-head-close"
+					>
+						<Gridicon icon="cross" size={ 16 } />
+					</Button>
+				</div>
+				<div className="developer-survey-notice__popup-img">
+					<img src={ surveyImage } alt={ translate( 'Code editor' ) } />
+				</div>
+				<div className="developer-survey-notice__popup-content">
+					<div className="developer-survey-notice__popup-content-title">
+						{ translate( 'Hey developer!' ) }
+					</div>
+					<div className="developer-survey-notice__popup-content-description">
+						{ translate(
+							"How do you use WordPress.com? Spare a moment? We'd love to hear what you think in a quick survey."
+						) }
+					</div>
+					<div className="developer-survey-notice__popup-content-buttons">
+						<Button
+							className="developer-survey-notice__popup-content-buttons-cancel"
+							onClick={ () => onClose( ONE_DAY_IN_SECONDS, 'remind-later-button' ) }
+						>
+							{ translate( 'Remind later' ) }
```

Comment:
This wording also struck me as odd while testing the page. To me, it should be `Remind _me_ later`. I know `Remind later` is the string used in the design, but it doesn't sound right IMO. cc @nightnei

Reply:
After discussion with ChatGPT - "Remind later" also totally right. 
Although the inclusion of "me" makes it more personal and engaging, there's a risk that its longer copy could inadvertently draw more attention than our main call-to-action.

Reply:
Looks like you are right. I don't mind to change. Personally for me, both sound good, but short option also sounds a bit better for me, but English is not my mother tongue, so probably we should ask @wongasy  for the final decision.

Reply:
@michaelpick We have a CTA `Remind later` that dismisses the developer survey prompt for one day. Typically we'd say `Remind me later`, but when it comes to a CTA, I wonder if omitting the object is also acceptable? 

<img width="466" alt="Screen Shot 2024-02-09 at 8 01 29 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/4d7817cf-b817-4fe2-a1c3-ad059c88fa38">

Reply:
`Maybe later` sounds like a great option to me! 👍